controllers/reconcilers/kisvc_route_reconciler.go

```diff
@@ -0,0 +1,206 @@
+	if err != nil {
+		return nil, err
+	}
+
+	disableIstioVirtualHost := ingressConfig.DisableIstioVirtualHost
+	if disableIstioVirtualHost == false {
+
+		serviceHost := getServiceHost(r.isvc)
+		isInternal := false
+		//if service is labelled with cluster local or knative domain is configured as internal
+		if val, ok := r.isvc.Labels[constants.VisibilityLabel]; ok && val == constants.ClusterLocalVisibility {
+			isInternal = true
+		}
+		serviceInternalHostName := network.GetServiceHostname(r.isvc.Name, r.isvc.Namespace)
+		if serviceHost == serviceInternalHostName {
+			isInternal = true
+		}
+		if isInternal {
+			return nil, nil
+		}
+
+		if ingressConfig.PathTemplate != "" {
+			serviceHost = ingressConfig.IngressDomain
+		}
+		annotations := utils.Filter(r.isvc.Annotations, func(key string) bool {
+			return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
+		})
+		route := &v1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        r.isvc.Name,
+				Namespace:   IstioNamespace,
+				Annotations: annotations,
+				Labels:      r.isvc.Labels,
+			},
+			Spec: v1.RouteSpec{
+				Host: serviceHost,
+				To: v1.RouteTargetReference{
+					Kind:   "Service",
+					Name:   IstioIngressService,
+					Weight: pointer.Int32(100),
+				},
+				Port: &v1.RoutePort{
+					TargetPort: intstr.FromString("https"),
+				},
+				TLS: &v1.TLSConfig{
+					Termination:                   v1.TLSTerminationPassthrough,
+					InsecureEdgeTerminationPolicy: v1.InsecureEdgeTerminationPolicyRedirect,
+				},
+				WildcardPolicy: v1.WildcardPolicyNone,
+			},
+			Status: v1.RouteStatus{
+				Ingress: []v1.RouteIngress{},
+			},
+		}
+		return route, nil
+	}
+	return nil, nil
+}
+
+func (r *kserveInferenceServiceRouteReconciler) getExistingResource() (*v1.Route, error) {
+	return r.routeHandler.FetchRoute(types.NamespacedName{Name: r.isvc.Name, Namespace: IstioNamespace})
+}
+
+func (r *kserveInferenceServiceRouteReconciler) processDelta(desiredRoute *v1.Route, existingRoute *v1.Route) (err error) {
+	comparator := r.routeHandler.GetComparator()
+	delta := r.deltaProcessor.ComputeDelta(comparator, desiredRoute, existingRoute)
+
+	if !delta.HasChanges() {
+		r.log.Info("No delta found")
+		return nil
+	}
+
+	if delta.IsAdded() {
+		r.log.Info("Will", "create", desiredRoute.GetName())
+		if err = r.client.Create(r.ctx, desiredRoute); err != nil {
+			return
+		}
+	}
+	if delta.IsUpdated() {
+		r.log.Info("Will", "update", existingRoute.GetName())
+		rp := existingRoute.DeepCopy()
+		rp.Labels = desiredRoute.Labels
+		rp.Annotations = desiredRoute.Annotations
+		rp.Spec = desiredRoute.Spec
+
+		if err = r.client.Update(r.ctx, rp); err != nil {
+			return
+		}
+	}
+	if delta.IsRemoved() {
+		r.log.Info("Will", "delete", existingRoute.GetName())
+		if err = r.client.Delete(r.ctx, existingRoute); err != nil {
+			return
+		}
+	}
+	return nil
+}
+
+func getServiceHost(isvc *kservev1beta1.InferenceService) string {
```

Comment:
This seems to be a copy of KServe's logic to set the top-level status.URL.
Can we use the URL that KServe already would set for us, rather than replicating code here?

Reply:
Actually in this function we are picking the URL that kserve already set in the status of InferenceService and then drive the host name from it.

Reply:
I was meaning that this code chooses between `status.components.predictor.url` or `status.components.transformer.url`. IMHO, it is better to simply look at `status.url`, because KServe has already set the needed one there, and you shouldn't need all this code.

Reply:
done